crates/extensions/c8y_mapper_ext/src/operations/handlers/custom_operation.rs

```diff
@@ -15,47 +20,108 @@ impl OperationContext {
         target: &EntityTarget,
         cmd_id: &str,
         message: &MqttMessage,
-        operation_name: &str,
-    ) -> Result<OperationOutcome, OperationError> {
+    ) -> (
+        Result<OperationOutcome, OperationError>,
+        Option<CumulocitySupportedOperations>,
+    ) {
```

Comment:
This return type is unusual.

- What's the rational?
- What does it means to return `Err(error), Some(operation))`?

Reply:
1. `Err` never returned actually, thus, I changed the return type in [064fd77](https://github.com/thin-edge/thin-edge.io/pull/3283/commits/064fd77cc8896951badf4260d290f1eda03f2272).
1. The c8y supported operation name has to be returned during processing a custom operation status transition. It can be `None`, if `.payload.c8y-mapper.on_fragment` is not provided.

I have to admit it is weird and the tuple return is a kind of a workaround for `OperationContext`...